applications/openshift/api-server/api_server_openshift_https_serving_cert/rule.yml

```
@@ -0,0 +1,50 @@
+documentation_complete: true
+
+prodtype: ocp4
+
+title: 'Make sure that the openshift-apiserver service uses TLS'
+
+description: |-
+    By default, the OpenShift API Server uses TLS. HTTPS should be
+    used for connections between openshift-apiserver and kube-apiserver.
+    OpenShift API server enables TLS automatically if a TLS key and a
+    certificate are provided via the <tt>serving-cert</tt> secret
+    in the <tt>openshift-apiserver</tt> namespace.
+
+rationale: |-
+    Connections between the kube-apiserver and the extension
+    openshift-apiserver could potentially carry sensitive data such
+    as secrets and keys. It is important to use in-transit encryption
+    for any communication between the kube-apiserver and the extension
+    openshift-apiserver.
+
+severity: medium
+
+references:
+    cis: 1.2.4
+
+ocil_clause: |-
+    Verify that the serving-cert for the openshift-apiserver is type
+    <tt>kubernetes.io/tls</tt> and that returned Data includes <tt>tls.crt</tt>
+    and <tt>tls.key</tt>.
```

The `ocil_clause` when fully built looks like:

```
Is it the case that Verify that the serving-cert for the openshift-apiserver
is type <tt>kubernetes.io/tls</tt> and that returned Data includes
<tt>tls.crt</tt> and <tt>tls.key</tt>?
```

which is a little weird, so would recommend something like:

```suggestion
    the openshift-apiserver serving-cert is not set to type
    <tt>kubernetes.io/tls</tt> and that returned Data does not include <tt>tls.crt</tt>
    and <tt>tls.key</tt>
```

---

Thank you, I used your suggestion. But I have two questions:

- are the `ocil_clause`s supposed to be written in a negative? I've seen some of them written in a positive ("...permissions are hardened...") and some of them in a negative ("...the permissions are open..."). Which is it?
- How do I view the ocil clauses rendered? To render the description and rationale I just view the built HTML guide (`build/guides/ssg-ocp4-guide-cis.html`) but I don't see the OCIL attributes there...

---

> Thank you, I used your suggestion. But I have two questions:
> 
> * are the `ocil_clause`s supposed to be written in a negative? I've seen some of them written in a positive ("...permissions are hardened...") and some of them in a negative ("...the permissions are open..."). Which is it?

There is no hard and fast rule as the answers to these questions are true/false. Generally, we have stuck to the negative more so than the positive. Basically, you are asking the auditor here to say "True, it is configured correctly" or "False, it isn't configured correctly".

> * How do I view the ocil clauses rendered? To render the description and rationale I just view the built HTML guide (`build/guides/ssg-ocp4-guide-cis.html`) but I don't see the OCIL attributes there...

Probably because the guides (purposefully) are XCCDF only. Currently, would have to check the OCIL file itself.